server/tests/Api.IntegrationTests/controllers/LecturesControllerTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Net;
+using System.Net.Http.Json;
+using Microsoft.AspNetCore.Mvc.Testing;
+using StudyZen.Application.Dtos;
+using StudyZen.Domain.Entities;
+
+namespace Api.IntegrationTests.controllers;
+
+public class LecturesControllerTests : IClassFixture<WebApplicationFactory<Program>>
+{
+    private readonly HttpClient _httpClient;
+    private readonly UpdateLectureDto _updateLectureDto;
+
+    public LecturesControllerTests(WebApplicationFactory<Program> factory)
+    {
+        _httpClient = factory.CreateClient();
+        _updateLectureDto = new UpdateLectureDto("New name", "New cont");
+    }
+
+    private async Task<CourseDto> CreateCourse()
+    {
+        var createCourseDto = new CreateCourseDto("Test name", "Test desc");
+        var courseResponse = await _httpClient.PostAsJsonAsync("Courses", createCourseDto);
+        Assert.Equal(HttpStatusCode.Created, courseResponse.StatusCode);
+        var course = await courseResponse.Content.ReadFromJsonAsync<CourseDto>();
+        Assert.NotNull(course);
+        return course;
+    }
+
+    [Fact]
+    public async Task CreateLecture()
+    {
+        var course = CreateCourse().Result;
```

Comment:
to avoid repeating the same lane in between tests, i would move data seeding to a separate method or at least do this in a SetUp method